src/group_theory/complement.lean

```diff
@@ -0,0 +1,303 @@
+  S ∈ right_transversals T ↔ ∀ g : G, ∃! s : S, g * (s : G)⁻¹ ∈ T :=
+begin
+  refine ⟨λ h g, _, λ h g, _⟩,
+  { obtain ⟨x, h1, h2⟩ := h g,
+    exact ⟨x.2, (congr_arg (∈ T) (eq_mul_inv_of_mul_eq h1)).mp x.1.2, λ y hy,
+      (prod.ext_iff.mp (h2 ⟨⟨g * y⁻¹, hy⟩, y⟩ (inv_mul_cancel_right g y))).2⟩ },
+  { obtain ⟨x, h1, h2⟩ := h g,
+    refine ⟨⟨⟨g * x⁻¹, h1⟩, x⟩, inv_mul_cancel_right g x, λ y hy, _⟩,
+    have := h2 y.2 ((congr_arg (∈ T) (eq_mul_inv_of_mul_eq hy)).mp y.1.2),
+    exact prod.ext (subtype.ext (eq_mul_inv_of_mul_eq ((congr_arg _ this).mp hy))) this },
+end
+
+@[to_additive] lemma mem_left_transverals_iff_exists_unique_quotient_mk'_eq :
+  S ∈ left_transversals (H : set G) ↔
+  ∀ q : quotient (quotient_group.left_rel H), ∃! s : S, quotient.mk' s.1 = q :=
+begin
+  have key : ∀ g h, quotient.mk' g = quotient.mk' h ↔ g⁻¹ * h ∈ H :=
+  @quotient.eq' G (quotient_group.left_rel H),
+  simp_rw [mem_left_transversals_iff_exists_unique_inv_mul_mem, set_like.mem_coe, ←key],
+  exact ⟨λ h q, quotient.induction_on' q h, λ h g, h (quotient.mk' g)⟩,
+end
+
+@[to_additive] lemma mem_right_transverals_iff_exists_unique_quotient_mk'_eq :
+  S ∈ right_transversals (H : set G) ↔
+  ∀ q : quotient (quotient_group.right_rel H), ∃! s : S, quotient.mk' s.1 = q :=
+begin
+  have key : ∀ g h, quotient.mk' g = quotient.mk' h ↔ h * g⁻¹ ∈ H :=
+  @quotient.eq' G (quotient_group.right_rel H),
+  simp_rw [mem_right_transversals_iff_exists_unique_mul_inv_mem, set_like.mem_coe, ←key],
+  exact ⟨λ h q, quotient.induction_on' q h, λ h g, h (quotient.mk' g)⟩,
+end
+
+@[to_additive] lemma mem_left_transverals_iff_bijective : S ∈ left_transversals (H : set G) ↔
+  function.bijective (S.restrict (quotient.mk' : G → quotient (quotient_group.left_rel H))) :=
+mem_left_transverals_iff_exists_unique_quotient_mk'_eq.trans
+  (function.bijective_iff_exists_unique (S.restrict quotient.mk')).symm
+
+@[to_additive] lemma mem_right_transverals_iff_bijective : S ∈ right_transversals (H : set G) ↔
+  function.bijective (set.restrict (quotient.mk' : G → quotient (quotient_group.right_rel H)) S) :=
+mem_right_transverals_iff_exists_unique_quotient_mk'_eq.trans
+  (function.bijective_iff_exists_unique (S.restrict quotient.mk')).symm
+
+@[to_additive] instance : inhabited (left_transversals (H : set G)) :=
+let hf  : function.has_right_inverse (quotient.mk' : G → quotient (quotient_group.left_rel H)) :=
+function.surjective.has_right_inverse quotient.surjective_quotient_mk', f := classical.some hf in
+⟨⟨set.range f, mem_left_transverals_iff_bijective.mpr ⟨by
+{ rintros ⟨_, q₁, rfl⟩ ⟨_, q₂, rfl⟩ hg,
+  rw [subtype.ext_iff, subtype.coe_mk, subtype.coe_mk,
+      ←classical.some_spec hf q₁, ←classical.some_spec hf q₂],
+  exact congr_arg f hg }, λ q, ⟨⟨f q, q, rfl⟩, classical.some_spec hf q⟩⟩⟩⟩
+
+@[to_additive] instance : inhabited (right_transversals (H : set G)) :=
+let hf  : function.has_right_inverse (quotient.mk' : G → quotient (quotient_group.right_rel H)) :=
```

Comment:
Isn't `quotient.out` the right inverse you're looking for here?

Reply:
Oh, of course. I'll change this tomorrow.